src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,67 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health;
+  name;
+
+  constructor(name = '', health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (this.health > 0) {
+      Animal.alive.push(this);
+    }
+  }
+
+  takeDamage(damage) {
+    this.health -= damage;
+
+    if (this.health <= 0) {
+      const index = Animal.alive.findIndex((animal) => animal === this);
+
+      Animal.alive.splice(index, 1);
```

Comment:
Instead of using findIndex and splice to remove the dead animal from the Animal.alive array, consider using filter for better performance. This will avoid the nested loop caused by findIndex and splice.